### DISCUSSION TITLE
Attachment to link

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#7415] [Help] [answered]
Hey,

I've tried it but I can't seem to get it work.
What I want is to upload a PDF or an image and display the link for the download of that attachment.
Is there a way to do that?

Sincerely,
Me

**Comment by melohagan** [comment]
Hi @kasgithub8 

Is this what you're looking for? 

Assume you have an attachment field on your table - in my case the User table:
![Screenshot 2022-08-23 at 17 20 04](https://user-images.githubusercontent.com/101575380/186210209-7d7574a9-eeea-49c5-b96e-a8eebebcbcc1.png)

Next add a link component with the URL binding set as follows:
`http://localhost:10000{{ Current User.File.0.url }}`
*Replace localhost:10000 with your domain*

The filename will be a random guid however. Not much you can do about that as far as I'm aware (it's the name of the file in the MinIO bucket).

![update_state](https://user-images.githubusercontent.com/101575380/186210849-92c0ab6a-5a5d-4a99-874f-4691a3a585ed.gif)

**Comment by kasgithub8** [comment, marked as answer]
@melohagan,
unfortunately no. I have a form with 3 Attachments.
I want to be able to get the link from those files in url fields. Reason is the size of the attachment fields is way too big and I dont want to add the attachment viewer in a table because it would be blowing things out of proportion.

**Reply by melohagan** [reply]
This should work for 3 attachments, you would just need three links and update the indices:
- `http://localhost:10000{{ Current User.File.0.url }}`
- `http://localhost:10000{{ Current User.File.1.url }}`
- `http://localhost:10000{{ Current User.File.2.url }}`

**Reply by kasgithub8** [reply]
But this is only for the current user right? I would need it per row for every user.

**Reply by melohagan** [reply]
The current user was just an example, but you can just as easily do this for any table. 

Here's another example:

**Data**
![Screenshot 2022-08-24 at 10 13 44](https://user-images.githubusercontent.com/101575380/186379922-f51fb73a-485f-42a8-b4c4-31434d1bbea2.png)

**Design**
![Screenshot 2022-08-24 at 10 15 54](https://user-images.githubusercontent.com/101575380/186380428-4d243618-37a8-4923-9075-9366120842a2.png)

**Result**
![update_state](https://user-images.githubusercontent.com/101575380/186381045-e441fc3f-e52f-43e7-9cfd-1aa686a838e1.gif)


*In this case because it's an image it wants to open in a new tab, but you can right click 'save link as'*

**App Export**
[Display Image Attachments-export-1661332736430.txt](https://github.com/Budibase/budibase/files/9414751/Display.Image.Attachments-export-1661332736430.txt)

**Reply by kasgithub8** [reply]
@melohagan 
http://localhost:10000{{ New Repeater.url }} 
doesn't work. Do I have to do something different?

**Reply by melohagan** [reply]
you need the index, e.g. `http://localhost:10000{{ New Repeater.NameOfAttachmentFieldHere.0.url }}`

**Reply by kasgithub8** [reply]
![Screenshot 2022-08-24 145810](https://user-images.githubusercontent.com/95284278/186424662-c23301bb-69ff-4910-a724-dffb3926c46f.png)
![Screenshot 2022-08-24 145925](https://user-images.githubusercontent.com/95284278/186424680-3b945acc-d3f2-4e62-9894-2042e53bff76.png)

**Reply by kasgithub8** [reply]
@melohagan 
this happens when I click a link inside your app: 
![image](https://user-images.githubusercontent.com/95284278/186432447-eef2b1d8-f74b-443d-ba6e-92f5b5542054.png)

**Reply by melohagan** [reply]
That's because the file is in the MinIO of my localhost. If you go to the data section you should be able to remove the existing missing files and add your own.

**Reply by kasgithub8** [reply]
I figured. I already tried it with my own pdf.
![image](https://user-images.githubusercontent.com/95284278/186436469-c4d63276-11ed-48e4-9913-bd159bc27bc2.png)

**Reply by melohagan** [reply]
Are you definitely hosting locally? You'll  need to replace `localhost:10000` if you're on cloud.

**Reply by kasgithub8** [reply]
> Are you definitely hosting locally? You'll need to replace `localhost:10000` if you're on cloud.

Yeah, we're hosting on a docker.

**Reply by kasgithub8** [reply]
Hey @melohagan, 

we found a little workaround.
Created an embed component with:
`const json = $("Form.Value");`
`var attachment = json['Item1File'][0]['url'];`
`return "<a href=" + attachment + ">Item File 1</a>"`

**Reply by melohagan** [reply]
Nice one - good job! 😄
